src/Smalot/PdfParser/Document.php

```diff
@@ -91,10 +91,26 @@ protected function buildDictionary()
         $this->dictionary = [];
 
         foreach ($this->objects as $id => $object) {
+            // Cache objects by type and subtype
             $type = $object->getHeader()->get('Type')->getContent();
 
-            if (!empty($type)) {
-                $this->dictionary[$type][$id] = $id;
+            if (null != $type) {
```

Comment:
Why changing from `!empty` to `null !=` here?

Reply:
Because the variable is declared here, thus `empty()` will not give anything more here than a null check. I suspect even a typed null check might work (or ideally the code should be made such). 

Using methods that check for more than necessary makes the code more confusing for others to read. Because it makes the reader think that all the functionality is necessary (the variable could be undeclared or the value equal to false/null -- and both checks are necessary for some reason when in reality they aren't).